### ISSUE TITLE
renderToReadableStream and Suspense runs async component twice

### ISSUE THREAD

**Issue by dmmikkel** [honojs/hono#1654]
### What version of Hono are you using?

next

### What runtime/platform is your app running on?

Deno

### What steps can reproduce the bug?

Use the following components

```jsx
const AsyncThing = async () => {
    console.log("Async start");
    await new Promise(resolve => setTimeout(resolve, 1000))
    console.log("Async end")
    return <div>Async thing</div>
}

export const Home = () => {
    console.log("Home")
    return <div>
        <h1>Hey på deg!</h1>
        <Suspense fallback={<div>Loading...</div>}>
            <AsyncThing />
        </Suspense>
    </div>
}
```

Then use renderToReadableStream on Home component

```typescript
const stream = renderToReadableStream(Home());
```

### What is the expected behavior?

I expect a console output like this

```terminal
Home
Async start
Async end
```

Indicating that the async component was ran only once.

### What do you see instead?

Instead I see

```terminal
Home
Async start
Async end
Async start
Async end
```

Indicating that the async component is ran twice.

### Additional information

When returning this as a stream the request takes ~2s to finish, instead of the expected ~1s.

**Comment by dmmikkel** [comment]
It seems to work as expected when changing

```typescript
        promise.then(async () => {
          return `<template>${await childrenToString(children)}</template><script>
```

to

```typescript
        promise.then(async (html) => {
          return `<template>${html}</template><script>
```

in jsx/streaming.ts

**Comment by usualoma** [comment]
Hi @dmmikkel!

Thank you for your report, we greatly appreciate you trying `Suspense`.

This issue will be fixed by #1656.

**Comment by dmmikkel** [comment]
np, and thanks for the quick fix.

**Comment by dmmikkel** [comment]
When testing it, it seems like it does not actually fix the problem.

Do you even need the `then`-part in `res = e.then(() => childrenToString(children))`? It worked without it for me.

**Comment by dmmikkel** [comment]
Nevermind, my mistake. Didn't refresh all dependencies in Deno.

**Comment by yusukebe** [comment]
Fixed with #1656.
